src/pudl/metadata/resources/mcoe.py

```diff
@@ -0,0 +1,224 @@
+"""Resource metadata for the MCOE (marginal cost of electricity) tables."""
+from typing import Any
+
+AGG_FREQS = ["yearly", "monthly"]
+
+RESOURCE_METADATA: dict[str, dict[str, Any]] = (
+    {
+        f"heat_rate_by_unit_{freq}": {
+            "description": (
+                f"{freq.title()} heat rate estimates by generation unit. Generation "
+                "units are identified by ``unit_id_pudl`` and are composed of a set of "
+                "interconnected boilers and generators."
+            ),
+            "schema": {
+                "fields": [
+                    "report_date",
+                    "plant_id_eia",
+                    "unit_id_pudl",
+                    "net_generation_mwh",
+                    "fuel_consumed_mmbtu",
+                    "heat_rate_mmbtu_mwh",
+                ],
+                "primary_key": [
+                    "report_date",
+                    "plant_id_eia",
+                    "unit_id_pudl",
+                ],
+            },
+            "field_namespace": "eia",
+            "sources": ["eia923", "eia860"],
+            "etl_group": "outputs",
+        }
+        for freq in AGG_FREQS
+    }
+    | {
+        f"heat_rate_by_generator_{freq}": {
+            "description": (
+                f"{freq.title()} heat rate estimates by generator. These are actually "
+                "just generation unit level heat rates, which have been broadcast "
+                "across all constituent generator IDs, since heat rates really only "
+                "have a well-defined meaning in the context of a generation unit."
+            ),
+            "schema": {
+                "fields": [
+                    "report_date",
+                    "plant_id_eia",
+                    "unit_id_pudl",
+                    "generator_id",
+                    "heat_rate_mmbtu_mwh",
+                    "fuel_type_code_pudl",
+                    "fuel_type_count",
+                ],
+                "primary_key": [
+                    "report_date",
+                    "plant_id_eia",
+                    "generator_id",
+                ],
+            },
+            "field_namespace": "eia",
+            "sources": ["eia923", "eia860"],
+            "etl_group": "outputs",
+        }
+        for freq in AGG_FREQS
+    }
+    | {
+        f"capacity_factor_by_generator_{freq}": {
+            "description": (
+                f"{freq.title()} estimates of generator capacity factor. Capacity "
+                "factor is calculated based on reported generator capacity and the "
+                "allocated net generation reported in the generation and generation "
+                "fuel tables."
```

Comment:
Similarly need to investigate, caveat and explain capacity factors that are < 0.0 and > 1.0.